src/System.Net.Primitives/src/System/Net/IPEndPoint.cs

```diff
@@ -145,12 +145,12 @@ public static bool TryParse(ReadOnlySpan<char> s, out IPEndPoint result)
 
             if (IPAddress.TryParse(s.Slice(0, addressLength), out IPAddress address))
             {
-                int port = 0;
+                uint port = 0;
                 if (addressLength == s.Length ||
-                    (int.TryParse(s.Slice(addressLength + 1), out port) && port >= MinPort && port <= MaxPort))
+                    (uint.TryParse(s.Slice(addressLength + 1), NumberStyles.None, CultureInfo.InvariantCulture, out port) && port >= MinPort && port <= MaxPort))
```

Comment:
You can then also remove the `&& port >= MinPort`, as that's guaranteed now by uint.